### PR TITLE
add new trait `GetDataBoxType`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add new trait
- Particles.hpp: add `GetDataBoxType` trait

### CODE REVIEW

File: src/picongpu/include/particles/Particles.hpp [hunk 1]
```diff
@@ -93,5 +93,14 @@ class Particles : public ParticlesBase<T_ParticleDescription, MappingDesc>, publ
     curandState* randState;
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/Particles.hpp]: see #727 - what is this?

File: src/picongpu/include/traits/GetDataBoxType.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * \tparam T_Type type from which you need the DataBoxType
+ * \treturn ::type
+ */
+template<typename T_Type>
+struct GetDataBoxType;
+
+} //namespace traits
+
+
+}// namespace picongpu
```
Comment (PrometheusPi) [inline src/picongpu/include/traits/GetDataBoxType.hpp]: please add newline

File: src/picongpu/include/particles/Particles.hpp (hunk 1 above)
Comment (psychocoderHPC) [inline src/picongpu/include/particles/Particles.hpp]: see #727^^

Comment (PrometheusPi) [inline src/picongpu/include/particles/Particles.hpp]: Thanks - will close this dialog.

File: src/picongpu/include/traits/GetDataBoxType.hpp (hunk 2 above)
Comment (psychocoderHPC) [inline src/picongpu/include/traits/GetDataBoxType.hpp]: fixed

**Comment by PrometheusPi** [comment]
I merged #727 - might cause some merge conflicts for this pull request.

**Pull request merged**